src/pyj/read_book/flow_mode.pyj

```diff
@@ -114,15 +119,19 @@ def flow_onwheel(evt):
 
 @check_for_scroll_end
 def goto_boundary(dir):
-    scroll_viewport.scroll_to(window.pageXOffset, 0 if dir is DIRECTION.Up else document_height())
+    scroll_viewport.scroll_to(scroll_viewport.x(), 0 if dir is DIRECTION.Up else document_height())
```

Comment:
It's important to use the value from ScrollViewport, again, so that the value goes positive from zero in both RTL and LTR books from the perspective of this code.